### ISSUE TITLE
false positive with using alternate values SC2068

### ISSUE THREAD

**Issue by vapier** [koalaman/shellcheck#669]
this code snippet is quoted correctly, but SC2068 is triggered anyways.  

``` sh
$ cat test.sh
echo ${@+"$@"}

$ shellcheck test.sh
Line 1:
echo ${@+"$@"}
     ^-- SC2068: Double quote array expansions to avoid re-splitting elements.
```

quoting on the tested var itself doesn't matter ... the quotes on the replacement will handle the expansion properly.

**Comment by vapier** [comment]
another example that might be more illustrative:
```sh
some-command --blah ${BUG+--bug "${BUG}" --include-bug-foo}
```

adding quotes around the outer layer would definitely be incorrect

**Comment by Corax26** [comment]
This one is really annoying indeed, especially since 1. there's no other simple way to conditionally pass arguments to commands and 2. disabling SC2068 globally is not an option, as it's one of the most critical checks.

**Comment by koalaman** [comment]
`:+` was already a handled case, and with 4243c6a `+` is as well. 

It's very easy to fool (it doesn't even check that the value is quoted), but it will at least stop warning here.

**Comment by Corax26** [comment]
Great, that was quick! Thank you @koalaman :)
